nrf-hal-common/src/twim.rs

```diff
@@ -333,6 +334,122 @@ where
         Ok(())
     }
 
+    /// Copy data into RAM and write to an I2C slave, then read data from the slave without
+    /// triggering a stop condition between the two
+    ///
+    /// The read buffer must have a length of at most 255 bytes on the nRF52832
+    /// and at most 65535 bytes on the nRF52840.
+    pub fn copy_write_then_read(
+        &mut self,
+        address: u8,
+        tx_buffer: &[u8],
+        rx_buffer: &mut [u8],
+    ) -> Result<(), Error> {
+        if rx_buffer.len() > EASY_DMA_SIZE {
+            return Err(Error::RxBufferTooLong);
+        }
+
+        // Conservative compiler fence to prevent optimizations that do not
+        // take in to account actions by DMA. The fence has been placed here,
+        // before any DMA action has started
+        compiler_fence(SeqCst);
+
+        self.0
+            .address
+            .write(|w| unsafe { w.address().bits(address) });
+
+        // Set up the DMA read
+        self.0.rxd.ptr.write(|w|
+            // We're giving the register a pointer to the stack. Since we're
+            // waiting for the I2C transaction to end before this stack pointer
+            // becomes invalid, there's nothing wrong here.
+            //
+            // The PTR field is a full 32 bits wide and accepts the full range
+            // of values.
+            unsafe { w.ptr().bits(rx_buffer.as_mut_ptr() as u32) });
+        self.0.rxd.maxcnt.write(|w|
+            // We're giving it the length of the buffer, so no danger of
+            // accessing invalid memory. We have verified that the length of the
+            // buffer fits in an `u8`, so the cast to the type of maxcnt
+            // is also fine.
+            //
+            // Note that that nrf52840 maxcnt is a wider
+            // type than a u8, so we use a `_` cast rather than a `u8` cast.
+            // The MAXCNT field is thus at least 8 bits wide and accepts the
+            // full range of values that fit in a `u8`.
+            unsafe { w.maxcnt().bits(rx_buffer.len() as _) });
+
+        // Chunk write data
+        let wr_buffer = &mut [0; FORCE_COPY_BUFFER_SIZE][..];
```

Comment:
```suggestion
        let wr_buffer = &mut [0; FORCE_COPY_BUFFER_SIZE.min(EASY_DMA_SIZE)][..];
```

I think we should take the minimum here too? Or ensure elseplace that `FORCE_COPY_BUFFER_SIZE` is always <= `EASY_DMA_SIZE`. Maybe we should do that where we define `FORCE_COPY_BUFFER_SIZE` even?

Reply:
Yes, i don't know how to do that min check in const though?
I guess ensuring `FORCE_COPY_BUFFER_SIZE <= EASY_DMA_SIZE` where we define `FORCE_COPY_BUFFER_SIZE` is the best solution, then we can just use `FORCE_COPY_BUFFER_SIZE` for our buffer and chunk sizes?

Reply:
Ok, I think we should use .min() at both places or at no place in this code then :) Otherwise it's just confusing I think.

Reply:
Totally agree, so i will use `FORCE_COPY_BUFFER_SIZE` in both places then :)

How can i assert `FORCE_COPY_BUFFER_SIZE <= EASY_DMA_SIZE` when they are defined in "top-level"?

Reply:
I am not sure we can ensure this at compile time other than manually making sure of this and maybe add a comment.

Reply:
The only hacky way i can think of right now is not pretty and not zero cost, something like:
`const _CHECK_FORCE_COPY_BUFFER_SIZE: usize = EASY_DMA_SIZE - FORCE_COPY_BUFFER_SIZE;`
`// ERROR: FORCE_COPY_BUFFER_SIZE must be <= EASY_DMA_SIZE`
..that would fail with "attempt to subtract with overflow" :)
But i guess just a comment should be enough right?

Reply:
Wont it just optimize that out and thus be zero cost?

Reply:
Ahh yes true!